runtime/src/main/java/io/novafoundation/nova/runtime/multiNetwork/connection/autobalance/NodeAutobalancer.kt

```diff
@@ -2,49 +2,41 @@ package io.novafoundation.nova.runtime.multiNetwork.connection.autobalance
 
 import android.util.Log
 import io.novafoundation.nova.common.utils.LOG_TAG
-import io.novafoundation.nova.runtime.ext.wssNodes
 import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
 import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
-import io.novafoundation.nova.runtime.multiNetwork.connection.ConnectionSecrets
 import io.novafoundation.nova.runtime.multiNetwork.connection.NodeWithSaturatedUrl
 import io.novafoundation.nova.runtime.multiNetwork.connection.autobalance.strategy.NodeSelectionStrategyProvider
-import io.novafoundation.nova.runtime.multiNetwork.connection.saturateNodeUrls
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.emitAll
-import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.transform
+import kotlinx.coroutines.flow.transformLatest
 
 class NodeAutobalancer(
     private val autobalanceStrategyProvider: NodeSelectionStrategyProvider,
-    private val connectionSecrets: ConnectionSecrets,
 ) {
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     fun connectionUrlFlow(
         chainId: ChainId,
         changeConnectionEventFlow: Flow<Unit>,
         availableNodesFlow: Flow<Chain.Nodes>,
     ): Flow<NodeWithSaturatedUrl?> {
-        return availableNodesFlow.flatMapLatest { nodesConfig ->
-            autobalanceStrategyProvider.strategyFlowFor(chainId, nodesConfig.nodeSelectionStrategy).transform { strategy ->
```

Comment:
No need to comlicate things with Flow here as we dont use it the way I originally expected